src/main/java/co/cask/hydrator/plugin/batch/HTTPSink.java

```diff
@@ -0,0 +1,402 @@
+        message = createFormMessage(input);
+        contentType = " application/x-www-form-urlencoded";
+      } else if (config.messageFormat.equals("Custom")) {
+        message = createCustomMessage(config.body, input);
+        contentType = "application/plain";
+      }
+      messages.append(message).append(config.delimiterForMessages);
+    }
+    StringTokenizer tokens = new StringTokenizer(messages.toString().trim(), config.delimiterForMessages);
+    if (config.batchSize == 1 || tokens.countTokens() == config.batchSize) {
+      do {
+        HttpURLConnection conn = null;
+        Map<String, String> headers = config.getRequestHeadersMap();
+        try {
+          URL url = new URL(config.url);
+          conn = (HttpURLConnection) url.openConnection();
+          if (conn instanceof HttpsURLConnection) {
+            //Disable SSLv3
+            System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2");
+            if (config.disableSSLValidation) {
+              disableSSLValidation();
+            }
+          }
+          conn.setRequestMethod(config.method.toUpperCase());
+          conn.setConnectTimeout(config.connectTimeout);
+          conn.setReadTimeout(config.readTimeout);
+          conn.setInstanceFollowRedirects(config.followRedirects);
+          conn.addRequestProperty("charset", config.charset);
+          for (Map.Entry<String, String> propertyEntry : headers.entrySet()) {
+            conn.addRequestProperty(propertyEntry.getKey(), propertyEntry.getValue());
+          }
+          if (config.method.equals("POST") || config.method.equals("PUT")) {
+            if (!headers.containsKey("Content-Type")) {
+              conn.addRequestProperty("Content-Type", contentType);
+            }
+          }
+
+          if (messages.length() > 0) {
+            conn.setDoOutput(true);
+            try (OutputStream outputStream = conn.getOutputStream()) {
+              LOG.info("Payload: " + messages.toString().trim());
+              outputStream.write(messages.toString().trim().getBytes(config.charset));
+            }
+          }
+          responseCode = conn.getResponseCode();
+          messages.setLength(0);
+          if (config.failOnNon200Response && responseCode != 200) {
+            exception = new IllegalStateException("Received error response. Response code: " + responseCode);
+          }
+          break;
+        } catch (MalformedURLException | ProtocolException e) {
+          throw new IllegalStateException("Error opening url connection. Reason: " + e.getMessage(), e);
+        } catch (Exception e) {
+          LOG.warn("Error making {} request to url {} with headers {}.", config.method, config.url, headers);
+          exception = e;
+        } finally {
+          if (conn != null) {
+            conn.disconnect();
+          }
+        }
+        retries++;
+      } while (retries < config.numRetries);
+      if (exception != null) {
+        throw exception;
+      }
+    }
```

Comment:
Please suggest one below:
Option1:We can override destroy() and if 'messages' is not empty then send the http request with the remaining messages.
Option2:We can take input from user on total number of records and use in transform() to send the http request for remaining records/messages.